cumulus/primitives/core/src/parachain_block_data.rs

```diff
@@ -0,0 +1,126 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// This file is part of Cumulus.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Provides [`ParachainBlockData`] and its historical versions.
+
+use alloc::vec::Vec;
+use codec::Encode;
+use sp_runtime::traits::Block as BlockT;
+use sp_trie::CompactProof;
+
+pub mod v0 {
+	use super::*;
+
+	#[derive(codec::Encode, codec::Decode, Clone)]
+	pub struct ParachainBlockData<B: BlockT> {
+		/// The header of the parachain block.
+		pub header: B::Header,
+		/// The extrinsics of the parachain block.
+		pub extrinsics: alloc::vec::Vec<B::Extrinsic>,
+		/// The data that is required to emulate the storage accesses executed by all extrinsics.
+		pub storage_proof: sp_trie::CompactProof,
+	}
+
+	impl<Block: BlockT> From<ParachainBlockData<Block>> for super::ParachainBlockData<Block> {
+		fn from(block_data: ParachainBlockData<Block>) -> Self {
+			Self::new(
+				alloc::vec![Block::new(block_data.header, block_data.extrinsics)],
+				block_data.storage_proof,
+			)
+		}
+	}
+}
+
+/// The parachain block that is created by a collator.
+///
+/// This is send as PoV (proof of validity block) to the relay-chain validators. There it will be
+/// passed to the parachain validation Wasm blob to be validated.
+#[derive(codec::Encode, codec::Decode, Clone)]
+pub enum ParachainBlockData<Block: BlockT> {
```

Comment:
nit: since it contains many blocks now, should we rename it?
```suggestion
pub enum ParachainBlocksData<Block: BlockT> {
```